File: js/indexOfAllCapitalLetters.js
// Problem# 225
// Create a function that takes a single string as argument and returns an ordered array containing the
// indices of all capital letters in the string.
// https://edabit.com/challenge/cRAqXCgyYnjdgktEq

const indexOfCaps = (str) => {
  const allCaps = str.match(/[A-Z]/g);

  return allCaps ? allCaps.map((letter) => str.indexOf(letter)) : [];
};

module.exports = indexOfCaps;

File: python/index_of_all_capital_letters.py
"""
Return the Index of All Capital Letters

Create a function that takes a single string as argument and returns an ordered list containing the indices of all capital letters in the string.

https://edabit.com/challenge/rQkriLJBc9CbfRbJb
"""
from typing import List


def index_of_caps(word: str) -> List[int]:
    return [i for i, letter in enumerate(word) if letter.isupper()]
